xtask/src/docgen.rs

```diff
@@ -48,6 +54,71 @@ pub fn typable_commands() -> Result<String, DynError> {
     Ok(md)
 }
 
+pub fn static_commands() -> Result<String, DynError> {
+    let mut md = String::new();
+    let keymap = EditorView::default().keymaps.map();
+    let keymaps = [
+        ("normal", keymap[&Mode::Normal].reverse_map()),
+        ("select", keymap[&Mode::Select].reverse_map()),
+        ("insert", keymap[&Mode::Insert].reverse_map()),
+    ];
+
+    md.push_str(&md_table_heading(&[
+        "Name".to_owned(),
+        "Description".to_owned(),
+        "Default keybinds".to_owned(),
+    ]));
+
+    for cmd in MappableCommand::STATIC_COMMAND_LIST {
+        let keymap_strings: Vec<String> = keymaps
+            .iter()
+            .enumerate()
+            .map(|(_, keymap)| {
+                keymap
+                    .1
+                    .get(cmd.name())
+                    .map(|bindings| {
+                        bindings.iter().fold(String::new(), |mut acc, bind| {
+                            if !acc.is_empty() {
+                                acc.push_str(", ");
+                            }
+                            acc.push_str("`` ");
+                            for key in bind {
+                                acc.push_str(&key.key_sequence_format());
+                            }
+                            acc.push_str(" ``");
+                            acc
+                        })
+                    })
+                    .unwrap_or_default()
+            })
+            .collect();
+
+        let keymap_string = keymap_strings
+            .iter()
+            .filter(|s| !s.is_empty())
+            .map(|s| {
+                let mode = match keymap_strings.iter().position(|t| t == s) {
+                    Some(0) => "normal",
+                    Some(1) => "select",
+                    Some(2) => "insert",
+                    _ => unreachable!(),
+                };
+                format!("{}: {}", mode, s)
+            })
+            .collect::<Vec<_>>()
+            .join(", ");
```

Comment:
Here you can remove the `enumerate` since the index isn't used and pass the `mode` part of the through:

```rust
let keymap_strings: Vec<_> = keymaps
    .iter()
    .map(|(mode, keymap)| {
        let bindings = keymap
            .get(cmd.name())
            .map(|bindings| {
                bindings.iter().fold(String::new(), |mut acc, bind| {
                    if !acc.is_empty() {
                        acc.push_str(", ");
                    }
                    acc.push_str("`` ");
                    for key in bind {
                        acc.push_str(&key.key_sequence_format());
                    }
                    acc.push_str(" ``");
                    acc
                })
            })
            .unwrap_or_default();

        (mode, bindings)
    })
    .collect();

let keymap_string = keymap_strings
    .iter()
    .filter(|(_, bindings)| !bindings.is_empty())
    .map(|(mode, bindings)| format!("{}: {}", mode, bindings))
    .collect::<Vec<_>>()
    .join(", ");
```